ipalib/util.py

```diff
@@ -520,10 +520,8 @@ def __get__(self, obj, cls):
         return self.store[obj]
 
     def __set__(self, obj, value):
-        raise AttributeError("can't set attribute")
+        self.store[obj] = value
```

Comment:
If you are going to turn the cached property into a writeable and deletable property, then you can remove the complicated weakref stuff and replace it with a simple non-data descriptor:

```
class cachedproperty(object):
    slots = ('fget',)
    def __init__(self, fget):
        self.fget = fget
    def __get__(self, obj, objtype=None):
        if obj is None:
            return self
        value = self.fget(obj)
        setattr(obj, self.fget.__name__, value)
        return value
```